### PR TITLE
Working with file project

### PR DESCRIPTION
**Pull request by ryby6969**

### CODE REVIEW

**Comment by halfaro1** [comment]
@ryby6969 great start. I believe you'll have a little more luck if you use the os.join function that was described in the textbook. Remember that although you may be running your program on one type of OS, the program needs to run on any type of OS.

**Comment by ryby6969** [comment]
I added the join function and still cannot get it to work. Not sure what I am doing wrong.

File: randomQuizGenerator.py [hunk 1]
```diff
@@ -28,8 +28,8 @@
             'Charleston', 'Wisconsin': 'Madison', 'Wyoming': 'Cheyenne'}
 #Generate 5 quiz files.
 for quizNum in range(5):
-    quizFile = open('./quizzes/capitalsquiz%s.txt' % (quizNum + 1), 'w')
-    answerKeyFile = open('.answers/capitalsquiz_answers%s.txt' % (quizNum + 1), 'w')
+    quizFile = open('./quizzes/capitalsquiz%s.txt' % (quizNum + 1),'w')
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @ryby6969 you should be using the `os.join` function any time you have to use slashes as part of a file path.

**Comment by halfaro1** [comment]
Closing and reopening the pull request to force the automated check to run...

**Comment by halfaro1** [comment]
@ryby6969 have you seen the results of the automated test?

**Comment by ryby6969** [comment]
I do not understand why it is not working. It still will not create the file folder for me.

File: randomQuizGenerator.py [hunk 2]
```diff
@@ -3,6 +3,10 @@
 # random order, along with the answer key.
 
 import random, os
+#Create folders \quizzes and \answers
+os.makedirs(os.path.join('./answers'), exist_ok=True)
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @ryby6969 try to stay away from using `./` use the `os.join` function whenever you need to use a slash

File: randomQuizGenerator.py [hunk 3]
```diff
@@ -22,12 +26,45 @@
             'Nashville', 'Texas': 'Austin', 'Utah': 'Salt Lake City', 'Vermont':
             'Montpelier', 'Virginia': 'Richmond', 'Washington': 'Olympia', 'West Virginia': 
             'Charleston', 'Wisconsin': 'Madison', 'Wyoming': 'Cheyenne'}
+#Generate 5 quiz files.
+for quizNum in range(5):
+    quizFile = open(os.path.join('quizzes/capitalsquiz%s.txt' % (quizNum + 1)),'w')
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @ryby6969 the proper usage of the `join` function here would be: `quizFile = open( os.path.join( 'quizzes', 'capitals quiz%s.txt' % (quizNum + 1) ), 'w')`

**Comment by ryby6969** [comment]
I am lost on this.

File: randomQuizGenerator.py [hunk 4]
```diff
@@ -3,6 +3,10 @@
 # random order, along with the answer key.
 
 import random, os
+#Create folders \quizzes and \answers
+os.makedirs(os.path.join('answer'), exist_ok=True)
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @ryby6969 you don't need to use `os.path.join` if you're referring to a single directory. So for lines 7 and 8, it's just a matter of: `os.makedirs( 'answers'  )`

File: randomQuizGenerator.py [hunk 5]
```diff
@@ -22,12 +26,45 @@
             'Nashville', 'Texas': 'Austin', 'Utah': 'Salt Lake City', 'Vermont':
             'Montpelier', 'Virginia': 'Richmond', 'Washington': 'Olympia', 'West Virginia': 
             'Charleston', 'Wisconsin': 'Madison', 'Wyoming': 'Cheyenne'}
+#Generate 5 quiz files.
+for quizNum in range(5):
+    quizFile = open(os.path.join('quizzes', 'capitalsquiz%s.txt') % (quizNum + 1),'w')
```
Comment (halfaro1) [inline randomQuizGenerator.py]: 👍

**Comment by halfaro1** [comment]
@ryby6969 don't give up! check out my suggestions above ☝️

File: randomQuizGenerator.py [hunk 6]
```diff
@@ -22,12 +26,45 @@
...
+    
+    quizFile.write('Name:\n\nDate:\n\nPeriod:\n\n')
+    quizFile.write((' ' * 20) + 'State Capitals Quiz (Form %s)' % (quizNum + 1))
+    quizFile.write('\n\n')
+
+    states = list(capitals.keys())
+    random.shuffle(states)
+
+    #Cycle through all 50 states.
+for questionNum in range(50):
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @ryby6969 shouldn't this be indented so it happens as a part of each quiz creation?

**Comment by halfaro1** [comment]
@ryby6969 
Fantastic work :+1:
I'm marking this pull request as closed, and labeling it as `graded` while I finish grading these projects. When I'm done grading all the projects, and your grade should be on Blackboard, I'll change the tag to `grade updated on blackboard`

**Pull request closed without merging**